src/context/AppContext.jsx

```diff
@@ -0,0 +1,29 @@
+import React, { createContext, useReducer, useEffect } from 'react';
+import { getVideos } from '../utils/api';
+
+export const Context = createContext({});
+
+const initialState = { searchTerm: '', videos: [] };
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'SEARCH_VIDEOS':
+      return { ...state, videos: action.payload };
+    default:
+      return state;
```

Comment:
Here I'd throw an error, there should not been calls to the dispatch function with unhandled actions